pkg/processor/schemaregistry/client_fake_test.go

```diff
@@ -0,0 +1,344 @@
+// Copyright © 2023 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build !integration
```

Comment:
It depends on the `integration` tag which file gets compiled, `client_fake_test.go` or `client_integration_test.go`. Both files contain an implementation of the `testSchemaRegistryURL` function, the one in this file spins up a fake schema registry server for test purposes and closes it after the test is done. The other simply returns the URL pointing to the schema registry running in docker (same as before the change).